src/main/java/com/exadel/discount/controller/CategoryController.java

```diff
@@ -0,0 +1,56 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.CategoryDTO;
+import com.exadel.discount.dto.validation.Create;
+import com.exadel.discount.service.CategoryService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@RequiredArgsConstructor
+@Validated
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @GetMapping
+    @ApiOperation("Get all categories")
+    public List<CategoryDTO> getAll() {
+        return categoryService.getAll();
+    }
+
+    @GetMapping("/{categoryId}")
+    @ApiOperation("Get information about category by Id")
+    public CategoryDTO getById(
+            @PathVariable(name = "categoryId") @NotNull UUID id) {
+        return categoryService.getById(id);
```

Comment:
.orElseThrow(new NotFoundException(String.format("Category with id %s not found", id));
maybe here, or something...